js/src/forum/components/UserBio.js

```diff
@@ -8,97 +8,113 @@ import extractText from 'flarum/utils/extractText';
  * edit it.
  */
 export default class UserBio extends Component {
-  init() {
+    init() {
+        /**
+         * Whether or not the bio is currently being edited.
+         *
+         * @type {Boolean}
+         */
+        this.editing = false;
+
+        /**
+         * Whether or not the bio is currently being saved.
+         *
+         * @type {Boolean}
+         */
+        this.loading = false;
+    }
+
+    view() {
+        const user = this.props.user;
+        const editable = this.props.user.attribute('canEditBio');
+        let content;
+
+        if (this.editing) {
+            content = (
+                <textarea
+                    className="FormControl"
+                    placeholder={extractText(app.translator.trans('fof-user-bio.forum.userbioPlaceholder'))}
+                    rows="3"
+                    value={user.bio()}
+                />
+            );
+        } else {
+            let subContent;
+
+            if (this.loading) {
+                subContent = <p className="UserBio-placeholder">{LoadingIndicator.component({ size: 'tiny' })}</p>;
+            } else {
+                const bioHtml = user.bioHtml();
+
+                if (bioHtml) {
+                    subContent = m.trust(bioHtml);
```

Comment:
Would using `m.trust` here open the _potential_ to place malicious code/any HTML in the bio content, especially without the `Formatter`?

Not sure, but thought it worth raising..

Reply:
I think it's safe. This is the code used to create `bioHtml()`.

```
User.prototype.bioHtml = computed('bio', bio =>
        bio
            ? '<p>' +
              $('<div/>')
                  .text(bio)
                  .html()
                  .replace(/\n/g, '<br>')
                  .autoLink({ rel: 'nofollow ugc' }) +
              '</p>'
            : ''
    );
```

To my knowledge `.text(text).html()` is safe against HTML injection.

I don't see any risk with the `replace` call either.

Dangerous point might be that `autoLink` library I have not reviewed.

I know some apps use a `template` element to convert text to html. But I'm not sure whether it has any benefit over this.

Reply:
Ok, great :)